tests/ui/coroutine/gen_block_is_no_future.stderr

```diff
@@ -1,10 +1,10 @@
-error[E0277]: `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:21}` is not a future
+error[E0277]: `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:8}` is not a future
   --> $DIR/gen_block_is_no_future.rs:4:13
    |
 LL | fn foo() -> impl std::future::Future {
-   |             ^^^^^^^^^^^^^^^^^^^^^^^^ `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:21}` is not a future
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^ `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:8}` is not a future
    |
-   = help: the trait `Future` is not implemented for `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:21}`
+   = help: the trait `Future` is not implemented for `{gen block@$DIR/gen_block_is_no_future.rs:5:5: 5:8}`
```

Comment:
We need to figure out how to point at the gen/async block in E0277s because the error as is isn't as clear as it should be: we point at the span that introduced the obligation, not where the obligation wasn't fulfilled.

Reply:
This is a general problem with opaques.

> we point at the span that introduced the obligation

This is the span that introduces the obligation. You're thinking about the span that introduces the expression that *assigns the hidden type of the opaque*. Determining what that span is is difficult to impossible without hacks that I would rather not try to land, because they will almost certainly regress with the new solver.